[2021-06-06 12:46:04,849] {taskinstance.py:851} INFO - Dependencies all met for <TaskInstance: forex_data_pipeline.spark1 2021-06-05T00:00:00+00:00 [queued]>
[2021-06-06 12:46:04,946] {taskinstance.py:851} INFO - Dependencies all met for <TaskInstance: forex_data_pipeline.spark1 2021-06-05T00:00:00+00:00 [queued]>
[2021-06-06 12:46:04,950] {taskinstance.py:1042} INFO - 
--------------------------------------------------------------------------------
[2021-06-06 12:46:04,952] {taskinstance.py:1043} INFO - Starting attempt 1 of 2
[2021-06-06 12:46:04,955] {taskinstance.py:1044} INFO - 
--------------------------------------------------------------------------------
[2021-06-06 12:46:04,992] {taskinstance.py:1063} INFO - Executing <Task(SparkSubmitOperator): spark1> on 2021-06-05T00:00:00+00:00
[2021-06-06 12:46:05,010] {standard_task_runner.py:52} INFO - Started process 160 to run task
[2021-06-06 12:46:05,040] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'forex_data_pipeline', 'spark1', '2021-06-05T00:00:00+00:00', '--job-id', '12', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/dag_spark.py', '--cfg-path', '/tmp/tmp6ha4lt21', '--error-file', '/tmp/tmp_bhksaq7']
[2021-06-06 12:46:05,051] {standard_task_runner.py:77} INFO - Job 12: Subtask spark1
[2021-06-06 12:46:05,264] {logging_mixin.py:104} INFO - Running <TaskInstance: forex_data_pipeline.spark1 2021-06-05T00:00:00+00:00 [running]> on host 54bd6187905d
[2021-06-06 12:46:05,542] {taskinstance.py:1257} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=admin@localhost.com
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=forex_data_pipeline
AIRFLOW_CTX_TASK_ID=spark1
AIRFLOW_CTX_EXECUTION_DATE=2021-06-05T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2021-06-05T00:00:00+00:00
[2021-06-06 12:46:05,596] {spark_submit.py:233} INFO - Could not load connection string spark_conn, defaulting to yarn
[2021-06-06 12:46:05,602] {spark_submit.py:364} INFO - Spark-Submit cmd: spark-submit --master yarn --name arrow-spark /opt/airflow/dags/test.py
[2021-06-06 12:46:06,128] {spark_submit.py:526} INFO - JAVA_HOME is not set
[2021-06-06 12:46:06,205] {taskinstance.py:1455} ERROR - Cannot execute: spark-submit --master yarn --name arrow-spark /opt/airflow/dags/test.py. Error code is: 1.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1112, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1285, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1315, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 183, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 455, in submit
    self._mask_cmd(spark_submit_cmd), returncode
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name arrow-spark /opt/airflow/dags/test.py. Error code is: 1.
[2021-06-06 12:46:06,242] {taskinstance.py:1503} INFO - Marking task as UP_FOR_RETRY. dag_id=forex_data_pipeline, task_id=spark1, execution_date=20210605T000000, start_date=20210606T124604, end_date=20210606T124606
[2021-06-06 12:46:06,391] {local_task_job.py:146} INFO - Task exited with return code 1
